docs/how-to/write-plans.md

```diff
@@ -1,173 +1,81 @@
-# Writing Bluesky plans for Blueapi
-
-
-For an introduction to bluesky plans and general forms/advice, `see the bluesky documentation <https://nsls-ii.github.io/bluesky/plans.html>`__. Blueapi has some additional requirements, which are explained below.
-
-## Plans
-
-While the bluesky project uses `plan` in a general sense to refer to any `Iterable` of `Msg`\ s which may be run by the `RunEngine`, blueapi distinguishes between a `plan` and a `stub`. This distinction is made to allow for a subset of `stub`\ s to be exposed and run, as `stub`\ s may not make sense to run alone.
-
-Generally, a `plan` includes at least one `open_run` and `close_run` and is a complete description of an experiment. If it does not, it is a `stub`. This distinction is made in the bluesky core library between the `plan`\ s and `plan_stub`\ s modules.
-
-## Type Annotations
-
-To be imported into the blueapi context, `plan`\ s and `stub`\ s must be the in the form of a `PlanGenerator`: any function that return a `MsgGenerator` (a python `Generator` that yields `Msg`\ s). `PlanGenerator` and `MsgGenerator` types are available to import from `dodal`.
-
-``` 
-   def foo() -> MsgGenerator:
-       # The minimum PlanGenerator acceptable to blueapi
-       yield from {}
-```
-
-> **_NOTE:_** `PlanGenerator` arguments must be annotated to enable blueapi to generate their schema
-
-**Input annotations should be as broad as possible**, the least specific implementation that is sufficient to accomplish the requirements of the plan.
-
-For example, if a plan is written to drive a specific implementation of Movable, but never calls any methods on the device and only yields bluesky `'set'` Msgs, it can be generalised to instead use the base protocol `Movable`.
-
-```
-   def move_to_each_position(axis: Movable) -> MsgGenerator:
-       # Originally written for SpecificImplementationMovable
-       for _ in range(i):
-           yield from abs_set(axis, location)
+# Write Bluesky Plans for Blueapi
+
+:::{seealso}
+[The bluesky documentation](https://blueskyproject.io/bluesky/main/index.html) for an introduction to bluesky plans and general forms/advice. Blueapi has some additional requirements, which are explained below.
+:::
+
+## Format
+
+:::{seealso}
+[Explanation of why blueapi treats plans in a special way](../explanations/plans.md)
+:::
+
+Plans in Python files look like this:
+
+```python 
+from bluesky.protocols import Readable, Movable
+from bluesky.utils import MsgGenerator
+from typing import Mapping, Any
+
+def my_plan(
+    detector: Readable, 
+    motor: Movable, 
+    steps: int, 
+    sample_name: str, 
+    extra_metadata: dict[str, Any]) -> MsgGenerator:
```

Comment:
nit: the plan linting in dodal enforces that plans have exactly this argument `metadata: dict[str, Any] | None = None` - would be nice to have this example compliant with that.

Reply:
See also line 43 of this file which says that plans should have this arg...